src/api/rest/v1/ds/controller.ts

```diff
@@ -39,6 +40,88 @@ export class DsController {
         }
     }
 
+    public async create(req: Request, res: Response) {
+        try {
+            const { context } = await Utils.getNetworkConnectionFromRequest(req)
+            const permissions = Utils.buildPermissions(req)
+            const schemaName = Utils.getSchemaFromParams(req.params.schema)
+
+            const ds = await context.openDatastore(schemaName, {
+                // @ts-ignore
+                permissions
+            })
+
+            const record = req.body.record
+            const options = req.body.options || {}
+            record.schema = schemaName
+            const result = await ds.save(record, options)
+
+            if (result) {
+                const savedRecord = await ds.get((<any> result).id, {})
+                res.json({
+                    success: true,
+                    record: savedRecord
+                })
+            } else {
+                res.json({
+                    success: false,
+                    errors: ds.errors
+                })
+            }
+        } catch (error) {
+            const message = error.message
+
+            res.status(500).send({
+                error: message
+            });
+        }
+    }
+
+    public async update(req: Request, res: Response) {
+        try {
+            const { context } = await Utils.getNetworkConnectionFromRequest(req)
+            const permissions = Utils.buildPermissions(req)
+            const schemaName = Utils.getSchemaFromParams(req.params.schema)
+            const rowId = req.params.recordId
+
+            const ds = await context.openDatastore(schemaName, {
+                // @ts-ignore
+                permissions
+            })
+
+            const record = req.body.record
+            record._id = rowId
+            record.schema = schemaName
+            const options = req.body.options || {}
+
+            const result = await ds.save(record, _.merge({}, options, {
+                forceInsert: true   // throws an error if the record already exists
```

Comment:
But we want it to exist when doing an update. Not sure I understand how the `forceInsert` works

Reply:
Oh, I messed up. Fetching it now.